packages/curve-ui-kit/src/shared/ui/LargeTokenInput.tsx

```diff
@@ -189,6 +189,9 @@ type Props<T> = DataType<T> & {
  * Calculate the new balance based on max balance and percentage, rounding to specified decimals
  */
 function calculateNewBalance<T extends Amount>(max: T, newPercentage: Decimal, balanceDecimals: number | undefined): T {
+  // Avoid loss of precision when clicking 'Max'
```

Comment:
does that happen even with `BigNumber` :skull:

Reply:
Well we're rounding below with `newBalance = BigNumber.min(newBalance.toFixed(balanceDecimals), max)`, which is fine for any percentage except 100%